android/pytorch_android/src/main/cpp/pytorch_jni.cpp

```diff
@@ -8,8 +8,83 @@
 
 #include <torch/script.h>
 
+#include <torch/csrc/autograd/record_function.h>
+
+#include "cmake_macros.h"
+
+
+#if defined(TRACE_ENABLED) && defined(__ANDROID__)
+#include <android/log.h>
+
+#include <android/trace.h>
+#include <dlfcn.h>
+
+#define ALOGI(...)                                                             \
+    __android_log_print(ANDROID_LOG_INFO, "pytorch-jni", __VA_ARGS__)
+#endif
+
 namespace pytorch_jni {
 
+#ifdef TRACE_ENABLED
```

Comment:
I would push this down to everywhere that you're currently doing `#ifdef __ANDROID__`.  That way, you don't have to put guards around all the Trace calls.  Optimized builds with tracing not enabled will just inline all the calls and you won't take any speed or size hit.